pkg/cmd/pulumi/main.go

```diff
@@ -61,6 +63,8 @@ func main() {
 
 	if err := NewPulumiCmd().Execute(); err != nil {
 		cmd.DisplayErrorMessage(err)
+		logging.Flush()
+		cmdutil.CloseTracing()
```

Comment:
The happy path does this here https://github.com/pulumi/pulumi/blob/469991f937e22b457beb57d9344e92799f6b81ee/pkg/cmd/pulumi/pulumi.go#L305

It also flushes out profile info. I left that out for now, as it needs the value of the `profiling` flag.